service/extensions/extensions.go

```diff
@@ -43,6 +44,7 @@ func (bes *Extensions) Start(ctx context.Context, host component.Host) error {
 		extLogger := components.ExtensionLogger(bes.telemetry.Logger, extID)
 		extLogger.Info("Extension is starting...")
 		if err := ext.Start(ctx, components.NewHostWrapper(host, extLogger)); err != nil {
+			extLogger.WithOptions(zap.AddStacktrace(zap.DPanicLevel)).Error("Failed to start extension", zap.Error(err))
```

Comment:
The reason we're adding the `AddStacktrace` option here, is to avoid adding the stacktrace, as it makes the log less readable and doesn't help much in most cases.

I haven't found a more idiomatic way of doing this with zap, and this looks a bit ugly.

Reply:
But on the line 45 we don't have it. Why do we need it here?

Reply:
Also, instead of logging it here, we should just enhance the error:

return fmt.Errorf("failed to start exporter %q: %w", extID.String(), err)

Reply:
> But on the line 45 we don't have it. Why do we need it here?

The logger is configured so errors add stacktraces, which I want to avoid here. I could also log an Info or Warning, but that feels like a lie. It's an error, I just don't want to put a stacktrace in the middle of the output from other components.

> Also, instead of logging it here, we should just enhance the error:
> 
> return fmt.Errorf("failed to start exporter %q: %w", extID.String(), err)

That will only enhance the error we see at the very end. The error we return here is ultimately packaged with errors from `.Shutdown()` and returned from `cobra.Command`. That'd still be an improvement, but in my opinion this way makes clear where the error happened, which helps new users in particular.

Reply:
I don't think we should "log" every error where it happens, instead we should return better errors that describe what happened. That is my philosophy in general, to avoid logging all over the place...

Reply:
How about logging the error from `.Start()` before we call `.Shutdown()`? Part of the problem here is that if we error while starting components, then we don't have telemetry set up, `.Shutdown()` returns its own (spurious) errors, and we end up with a monster like this:
```
2023/01/31 18:40:52 collector server run finished with error: cannot start pipelines: failed to resolve authenticator "not_there": authenticator not found; failed to shutdown pipelines: no existing monitoring routine is running; no existing monitoring routine is running
```

> I don't think we should "log" every error where it happens, instead we should return better errors that describe what happened. That is my philosophy in general, to avoid logging all over the place...

I don't think it needs to happen in the same place *in the code*, but I do think that if we want to log an error, we should do it before we start logging other unrelated information. My problem with this situation is that it proceeds as follows:

1. Get an error during component start
2. Shutdown, producing more log output unrelated to the error above
3. Log the component start error along with other errors encountered during shutdown